scripts/generate-images.js

```diff
@@ -174,6 +180,16 @@ class GenerateImages {
                 await this.processSvg(filePath, filename, lessFile);
             }),
         );
+        if (defsList.length > 0) {
+            const defsEl = this.masterDocument.createElement("defs");
```

Comment:
We need to have it wrapped in a defs tag. A document fragment can't have that from what i understand 
Plus this is only in the script when we generate icons

Reply:
Couldn't you create the `defs` tag and a document fragment that appends all the defs, and then append it only once inside of the `defs`? That would definitely be more performant. It's not an issue now because it's a small sample size, but the script will gradually get slower with additional `defs`. If we do it with fragments, it will scale with 0 additional overhead with more gradient icons.